quarkchain/cluster/jsonrpc.py

```diff
@@ -1428,5 +1420,44 @@ def shutdown(self):
         return "pong"
 
     @public_methods.add
-    async def echo(self, params):
-        return params
+    async def qkc_subscribe(self, sub_type, full_shard_id, context):
+        if context is None or full_shard_id is None:
+            return None
+        self.sub_id += 1
+        response = {"jsonrpc": "2.0", "result": self.sub_id, "id": 1}  # TODO: change id
+        await context.send(str(response))
+
+        full_shard_id = shard_id_decoder(full_shard_id)
+        branch = Branch(full_shard_id)
+        shard = self.slave.shards.get(branch, None)
+        if not shard:
```

Comment:
non-exist should close the connection. does returning none close the connection?